src/include/storage/write_ahead_log/log_manager.h

```diff
@@ -1,23 +1,31 @@
 #pragma once
 
+#include <memory>
 #include <queue>
 #include <string>
 #include <unordered_map>
 #include <utility>
 #include <vector>
+#include "common/container/concurrent_blocking_queue.h"
+#include "common/dedicated_thread_owner.h"
+#include "common/managed_pointer.h"
 #include "common/spin_latch.h"
 #include "common/strong_typedef.h"
 #include "storage/record_buffer.h"
+#include "storage/write_ahead_log/log_consumer.h"
 #include "storage/write_ahead_log/log_io.h"
 #include "storage/write_ahead_log/log_record.h"
 #include "transaction/transaction_defs.h"
 
+// TODO(Utkarsh): Get rid of magic constants
+#define MAX_BUF 2
+
 namespace terrier::storage {
 /**
  * A LogManager is responsible for serializing log records out and keeping track of whether changes from a transaction
  * are persistent.
  */
-class LogManager {
+class LogManager : public DedicatedThreadOwner {
```

Comment:
The log manager probably does not own the `LogConsumerTask`. The ownership should be left for a more upper level component, maybe? Let's talk about this in a team meeting maybe.